### ISSUE TITLE
Stop using deprecated methods

### ISSUE THREAD

**Issue by jamesaoverton** [ontodev/robot#408]
Running `mvn package -Dmaven.compiler.showDeprecation=true` results in the following warnings (and lots of other noise):

```
[WARNING] robot-core/src/main/java/org/obolibrary/robot/ExtractOperation.java:[43,36] getEntitiesInSignature(org.semanticweb.owlapi.model.IRI,boolean) in org.semanticweb.owlapi.model.OWLSignatureBooleanArgs has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/ReportOperation.java:[456,40] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/OntologyHelper.java:[1245,15] OWLOntologyID(org.semanticweb.owlapi.model.IRI) in org.semanticweb.owlapi.model.OWLOntologyID has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/OntologyHelper.java:[1247,15] OWLOntologyID(org.semanticweb.owlapi.model.IRI,org.semanticweb.owlapi.model.IRI) in org.semanticweb.owlapi.model.OWLOntologyID has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[132,26] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[143,40] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[167,36] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[428,28] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[460,41] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:[533,36] create(org.apache.jena.sparql.core.DatasetGraph) in org.apache.jena.query.DatasetFactory has been deprecated
[WARNING] robot-core/src/main/java/org/obolibrary/robot/IOHelper.java: /Users/james/Repositories/github/ontodev/robot/robot-core/src/main/java/org/obolibrary/robot/IOHelper.java uses unchecked or unsafe operations.
```

**Comment by jamesaoverton** [comment]
I think that #412 covers everything I listed here, but when I run `mvn package -Dmaven.compiler.showDeprecation=true` I see more, different warnings:

```
[WARNING] robot-core/src/test/java/org/obolibrary/robot/CoreTest.java:[71,12] addIRIMapper(org.semanticweb.owlapi.model.OWLOntologyIRIMapper) in org.semanticweb.owlapi.model.OWLOntologyManager has been deprecated
[WARNING] robot-core/src/test/java/org/obolibrary/robot/ReasonOperationTest.java:[76,77] org.semanticweb.HermiT.Reasoner.ReasonerFactory in org.semanticweb.HermiT.Reasoner has been deprecated
[WARNING] robot-core/src/test/java/org/obolibrary/robot/ReasonOperationTest.java:[114,77] org.semanticweb.HermiT.Reasoner.ReasonerFactory in org.semanticweb.HermiT.Reasoner has been deprecated
[WARNING] robot-core/src/test/java/org/obolibrary/robot/ReasonOperationTest.java:[138,77] org.semanticweb.HermiT.Reasoner.ReasonerFactory in org.semanticweb.HermiT.Reasoner has been deprecated
[WARNING] robot-core/src/test/java/org/obolibrary/robot/ReasonOperationTest.java:[166,77] org.semanticweb.HermiT.Reasoner.ReasonerFactory in org.semanticweb.HermiT.Reasoner has been deprecated
[WARNING] robot-core/src/test/java/org/obolibrary/robot/RelatedObjectsHelperTest.java: robot-core/src/test/java/org/obolibrary/robot/RelatedObjectsHelperTest.java uses unchecked or unsafe operations.
[WARNING] robot-core/src/test/java/org/obolibrary/robot/RelatedObjectsHelperTest.java: Recompile with -Xlint:unchecked for details.
```

**Comment by balhoff** [comment]
I think it should be `org.semanticweb.HermiT.ReasonerFactory`.

**Comment by rctauber** [comment]
I replaced those test methods as well, but now I'm seeing a whole new set of warnings:
```
[WARNING] commons-logging-1.1.3.jar, jcl-over-slf4j-1.7.22.jar define 6 overlapping classes: 
[WARNING]   - org.apache.commons.logging.impl.SimpleLog$1
[WARNING]   - org.apache.commons.logging.Log
[WARNING]   - org.apache.commons.logging.impl.SimpleLog
[WARNING]   - org.apache.commons.logging.LogConfigurationException
[WARNING]   - org.apache.commons.logging.impl.NoOpLog
[WARNING]   - org.apache.commons.logging.LogFactory
[WARNING] owlapi-distribution-4.5.6.jar, owlapi-tools-4.5.6.jar define 83 overlapping classes: 
```
(and so on, for many lines)

**Comment by jamesaoverton** [comment]
Yes, I'm not worried about the overlapping classes warnings yet.

**Comment by rctauber** [comment]
Alright, I just pushed those fixes to #412

**Comment by jamesaoverton** [comment]
Closed by #412
